pkg/api/types.go

```diff
@@ -341,14 +342,101 @@ type Result struct {
 	BuildInfo BuildInfo
 }
 
-// BuildInfo holds information about a particular step in the build process.
+// BuildInfo contains information about the build process.
 type BuildInfo struct {
+	// Stages contains details about each build stage.
+	Stages Stages
+
 	// FailureReason is a camel case reason that is used by the machine to reply
 	// back to the OpenShift builder with information why any of the steps in the
-	// build, failed.
+	// build failed.
 	FailureReason FailureReason
 }
 
+// Stages is a slice of build stages that have been recorded.
+type Stages []StageInfo
+
+// StageInfo contains details about a build stage.
+type StageInfo struct {
+	// StageName is the identifier for each build stage.
+	StageName StageName
+
+	// StartTime identifies when this stage started.
+	StartTime time.Time
+
+	// EndTime identifies when this stage ended.
+	EndTime time.Time
+
+	// Duration identifies how long this stage ran.
+	Duration time.Duration
```

Comment:
we shouldn't need both duration and endtime (if you have one, you can get the other), since ultimately we're targeting an api of "starttime" and "duration", just store those two things.